plugin/src/main/java/org/opensearch/ml/processor/ModelExecutor.java

```diff
@@ -334,4 +341,26 @@ default List<String> writeNewDotPathForNestedObject(Object json, String dotPath)
     default String convertToDotPath(String path) {
         return path.replaceAll("\\[(\\d+)\\]", "$1\\.").replaceAll("\\['(.*?)']", "$1\\.").replaceAll("^\\$", "").replaceAll("\\.$", "");
     }
+
+    default void setRequestContextFromExt(SearchRequest request, PipelineProcessingContext requestContext) {
+
+        List<SearchExtBuilder> extBuilderList = request.source().ext();
+        for (SearchExtBuilder ext : extBuilderList) {
+            if (ext instanceof MLInferenceRequestParametersExtBuilder) {
+                MLInferenceRequestParametersExtBuilder mlExtBuilder = (MLInferenceRequestParametersExtBuilder) ext;
+                Map<String, Object> mlParams = mlExtBuilder.getRequestParameters().getParams();
+                mlParams
+                    .forEach(
+                        (key, value) -> requestContext
+                            .setAttribute(String.format("ext.%s.%s", MLInferenceRequestParametersExtBuilder.NAME, key), value)
```

Comment:
you need to use locale with `String.format`, default should be good enough

Reply:
agree